tools/nl/eval/main.py

```diff
@@ -0,0 +1,111 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import json
+import os
+
+from absl import app
+from absl import flags
+from google.cloud import aiplatform
+from google.cloud import aiplatform_v1
+import ndcg
+import numpy as np
+import yaml
+
+PROJECT = 'datcom-website-dev'
+LOCATION = 'us-central1'
+VECTOR_SERACH_ENDPOINT = "302175072.us-central1-496370955550.vdb.vertexai.goog"
+INDEX_ENDPOINT = "projects/496370955550/locations/us-central1/indexEndpoints/8500794985312944128"
+NEIGHBOR_COUNT = 30
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    'model_name', '',
+    'Model name used for eval. Full list can be found in endpoints.yaml')
+
+flags.DEFINE_string(
+    'eval_folder', 'base',
+    'The folder for a eval unit. It should contain a golden.json file')
+
+
+def _load_models():
+  with open('vertex_ai_endpoints.yaml') as f:
+    return yaml.full_load(f)
+
+
+def _load_baseline():
+  with open(os.path.join(FLAGS.eval_folder, 'golden.json')) as f:
+    return json.load(f)
+
+
+def main(_):
+  aiplatform.init(project=PROJECT, location=LOCATION)
+  base_line = _load_baseline()
+  models = _load_models()
+  if FLAGS.model_name not in models:
+    print('Model not found')
+    return
+  model_info = models.get(FLAGS.model_name)
+  prediction_endpoint = aiplatform.Endpoint(
+      model_info['prediction_endpoint_id'])
+  vector_search_client = aiplatform_v1.MatchServiceClient(
+      client_options={"api_endpoint": VECTOR_SERACH_ENDPOINT},)
+
+  debug = {}
+  report = []
+  for query, base_line_matches in base_line.items():
+    print(query)
+    query_vector = prediction_endpoint.predict(instances=[query]).predictions[0]
+    query_norm = np.linalg.norm(query_vector)
+    vector_datapoint = aiplatform_v1.IndexDatapoint(feature_vector=query_vector)
+    vector_search_query = aiplatform_v1.FindNeighborsRequest.Query(
+        datapoint=vector_datapoint, neighbor_count=NEIGHBOR_COUNT)
+    vector_search_req = aiplatform_v1.FindNeighborsRequest(
+        index_endpoint=INDEX_ENDPOINT,
+        deployed_index_id=model_info['index_id'],
+        queries=[vector_search_query],
+        return_full_datapoint=True,
+    )
+    vector_search_resp = vector_search_client.find_neighbors(vector_search_req)
+    ranked_stat_vars = []
+    debug[query] = {'vector_search': []}
+    for n in vector_search_resp.nearest_neighbors[0].neighbors:
+      dp = n.datapoint
+      stat_var = dp.restricts[0].allow_list[0]
+      if stat_var not in ranked_stat_vars:
+        ranked_stat_vars.append(stat_var)
+      dp_norm = np.linalg.norm(dp.feature_vector)
+      debug[query]['vector_search'].append({
+          'sentence': dp.datapoint_id,
+          'stat_var': dp.restricts[0].allow_list[0],
+          'distance': n.distance / dp_norm / query_norm
+      })
+    if len(ranked_stat_vars) > len(base_line_matches):
+      ranked_stat_vars = ranked_stat_vars[:len(base_line_matches)]
```

Comment:
Wonder why we truncate to base-line match count?  if the ranked-svs have some svs lower ranked, is that useful to know?

Reply:
It's a good point. I will look into this later and see if it makes sense to include more (and how many)